Classes/Format/Mods.php

```diff
@@ -78,56 +119,216 @@ public function extractMetadata(\SimpleXMLElement $xml, array &$metadata)
-                    $metadata['place_sorting'][0] = preg_replace('/[[:punct:]]/', '', (string) $place);
+                $this->metadata['place'][] = (string) $place;
+                if (!$this->metadata['place_sorting'][0]) {
+                    $this->metadata['place_sorting'][0] = preg_replace('/[[:punct:]]/', '', (string) $place);
                 }
             }
         }
+    }
+
+    /**
+     * Get "year" and "year_sorting".
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getYears() {
         // Get "year_sorting".
-        if (($years_sorting = $xml->xpath('./mods:originInfo[not(./mods:edition="[Electronic ed.]")]/mods:dateOther[@type="order" and @encoding="w3cdtf"]'))) {
+        if (($years_sorting = $this->xml->xpath('./mods:originInfo[not(./mods:edition="[Electronic ed.]")]/mods:dateOther[@type="order" and @encoding="w3cdtf"]'))) {
             foreach ($years_sorting as $year_sorting) {
-                $metadata['year_sorting'][0] = intval($year_sorting);
+                $this->metadata['year_sorting'][0] = intval($year_sorting);
             }
         }
         // Get "year" and "year_sorting" if not specified separately.
-        $years = $xml->xpath('./mods:originInfo[not(./mods:edition="[Electronic ed.]")]/mods:dateIssued[@keyDate="yes"]');
+        $years = $this->xml->xpath('./mods:originInfo[not(./mods:edition="[Electronic ed.]")]/mods:dateIssued[@keyDate="yes"]');
         // Get "year" and "year_sorting" again if that was to sophisticated.
         if (empty($years)) {
             // Get all dates and assume these are dates of publication.
-            $years = $xml->xpath('./mods:originInfo/mods:dateIssued');
+            $years = $this->xml->xpath('./mods:originInfo/mods:dateIssued');
         }
         if (!empty($years)) {
             foreach ($years as $year) {
-                $metadata['year'][] = (string) $year;
-                if (!$metadata['year_sorting'][0]) {
+                $this->metadata['year'][] = (string) $year;
+                if (!$this->metadata['year_sorting'][0]) {
                     $year_sorting = str_ireplace('x', '5', preg_replace('/[^\d.x]/i', '', (string) $year));
                     if (
                         strpos($year_sorting, '.')
                         || strlen($year_sorting) < 3
                     ) {
                         $year_sorting = ((intval(trim($year_sorting, '.')) - 1) * 100) + 50;
                     }
-                    $metadata['year_sorting'][0] = intval($year_sorting);
+                    $this->metadata['year_sorting'][0] = intval($year_sorting);
                 }
             }
         }
     }
+
+    /**
+     * Get "description" stored in recordInfoNote.
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getDescription() {
+        $this->getSingleMetadata('description', './mods:recordInfo/mods:recordInfoNote/text()');
+    }
```

Comment:
An implementation in this class is only necessary for metadata which can't be queried by XPath alone (like for example author or holder, where multiple possible encodings have to be taken in account). Everything else should be configured in the backend (with defaults in `Resources/Private/Data/MetadataDefaults.php`).

Reply:
Thanks for info! I will make changes according to it in this PR.